src/poetry/core/version/markers.py

```diff
@@ -725,6 +743,7 @@ def is_empty(self) -> bool:
         return all(m.is_empty() for m in self._markers)
 
 
+@functools.lru_cache(maxsize=None)
```

Comment:
How many markers do we have for an average installation?

Only thinking if wouldn't make sense to set a boundary to the cache here.

Reply:
> How many markers do we have for an average installation?

Interesting question, but difficult to say. I tested some projects, but it's subjective if they are typical.

```
CacheInfo(hits=16070, misses=25, maxsize=None, currsize=25)
CacheInfo(hits=6369, misses=36, maxsize=None, currsize=36)
CacheInfo(hits=55, misses=14, maxsize=None, currsize=14)
CacheInfo(hits=289, misses=25, maxsize=None, currsize=25)
CacheInfo(hits=17518, misses=48, maxsize=None, currsize=48)
CacheInfo(hits=1124, misses=45, maxsize=None, currsize=45)
CacheInfo(hits=67168, misses=60, maxsize=None, currsize=60)
CacheInfo(hits=66985, misses=78, maxsize=None, currsize=78)
```

`currsize` is the number of unique markers. `hits + misses` is the number of all parsed markers.

> Only thinking if wouldn't make sense to set a boundary to the cache here.

I don't think it's necessary. On further testing I could even observe a reduction of peak memory usage. I suppose this is because now we only create one object per unique marker, before we created several ones (and probably held a lot of them until the end).